config/config.go

```diff
@@ -1721,7 +1722,46 @@ type Image struct {
 	Tag string `json:"tag,omitempty"`
 }
 
-func (i Image) String() (ref string) {
+// UnmarshalJSON makes the schema change from string to Image backwards compatible
+func (i *Image) UnmarshalJSON(data []byte) error {
+	var str string
+	if err := json.Unmarshal(data, &str); err == nil {
+		ParseImageRef(str, i)
+		return nil
+	}
+
+	type Alias Image // Prevent infinite recursion
+	aux := &struct {
+		*Alias
+	}{
+		Alias: (*Alias)(i),
+	}
+
+	return json.Unmarshal(data, &aux)
```

Comment:
Is there an advantage to the embedded struct pointer vs this?
```golang
type Alias Image
var aux Alias
err := json.Unmarshal(data, &aux)
if err != nil {
    return err
}
*i = Image(aux)
return nil
```

Reply:
Don't see really a meaningful difference.
It's maybe a bit more explicit.